src/test/Microsoft.DotNet.CoreSetup.Packaging.Tests/NuGetArtifactTester.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.CoreSetup.Test;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Xml.Linq;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Packaging.Tests
+{
+    public class NuGetArtifactTester : IDisposable
+    {
+        public static NuGetArtifactTester Open(
+            RepoDirectoriesProvider dirs,
+            string project,
+            string id = null)
+        {
+            var tester = OpenOrNull(dirs, project, id);
+            Assert.NotNull(tester);
+            return tester;
+        }
+
+        public static NuGetArtifactTester OpenOrNull(
+            RepoDirectoriesProvider dirs,
+            string project,
+            string id = null)
+        {
+            id = id ?? project;
+
+            string nuspecPath = Path.Combine(
+                dirs.BaseBinFolder,
+                project,
+                dirs.Configuration,
+                "specs",
+                $"{id}.nuspec");
+
+            if (!File.Exists(nuspecPath))
+            {
+                return null;
+            }
+
+            PackageIdentity builtIdentity = new NuspecReader(nuspecPath).GetIdentity();
+
+            string path = Path.Combine(
```

Comment:
(nit) from a readability perspective, I think naming this nupkgPath would help since there is already another `path` related variable previously declared.